src/client/GameWindow.cpp

```diff
@@ -140,122 +140,148 @@ namespace client
+            *clickMode = false;
 
-                    if (moveMode && clickMode)
-                    {
+            break;
 
-                        newMapOffset = {sf::Mouse::getPosition(*clientGameWindow).x - clickStartingPoint.x,
-                                        sf::Mouse::getPosition(*clientGameWindow).y - clickStartingPoint.y};
+        case sf::Event::MouseMoved:
 
-                        firstHexagonPosition = {firstHexagonPosition[0] + newMapOffset[0],
-                                                firstHexagonPosition[1] + newMapOffset[1]};
+            if (*moveMode && *clickMode)
+            {
 
-                        for (unsigned i = 0; i < mapTextureToDisplay.size(); i++)
-                            mapTextureToDisplay[i].moveSpritePosition(newMapOffset[0], newMapOffset[1]);
+                newMapOffset = { sf::Mouse::getPosition(*clientGameWindow).x - clickStartingPoint->x,
+                                 sf::Mouse::getPosition(*clientGameWindow).y - clickStartingPoint->y};
 
-                        for (auto &kv : elementTextureToDisplay)
-                        {
-                            kv.second->moveSpritePosition(newMapOffset[0], newMapOffset[1]);
-                        }
+                firstHexagonPosition = {firstHexagonPosition[0] + newMapOffset[0],
+                                        firstHexagonPosition[1] + newMapOffset[1]};
 
-                        clickStartingPoint = sf::Mouse::getPosition(*clientGameWindow);
-                    }
-                    break;
+                for (unsigned i = 0; i < mapTextureToDisplay.size(); i++)
+                {
+                    mapTextureToDisplay[i].moveSpritePosition(newMapOffset[0], newMapOffset[1]);
+                }
 
-                case sf::Event::KeyPressed:
+                for (auto &kv : elementTextureToDisplay)
+                {
+                    kv.second->moveSpritePosition(newMapOffset[0], newMapOffset[1]);
+                }
 
-                    switch (event.key.code)
-                    {
-                    case sf::Keyboard::M:
+                *clickStartingPoint = sf::Mouse::getPosition(*clientGameWindow);
+            }
+            break;
 
-                        if (moveMode)
-                        {
-                            moveMode = false;
-                            if (clientCursor.loadFromSystem(sf::Cursor::Arrow))
-                                clientGameWindow->setMouseCursor(clientCursor);
-                        }
-                        else
-                        {
-                            moveMode = true;
-                            if (clientCursor.loadFromSystem(sf::Cursor::Hand))
-                                clientGameWindow->setMouseCursor(clientCursor);
-                        }
-                        break;
+        case sf::Event::KeyPressed:
 
-                    case sf::Keyboard::K:
+            switch (event.key.code)
```

Comment:
not very clean and complicated to read

Reply:
(about switch nesting)

Reply:
The 2nd switch will desapear when the button will be ready. The keyboard event will be catch by sf::Event::KeyPressed: and it will be easily to understand. The actual solution is temporary (and here since 2 months)